kitsune/wiki/services.py

```diff
@@ -80,42 +93,110 @@ def find_stale_translation_candidates(
         ]
         return candidates
 
-    def process_stale_translations(
-        self, limit: int | None = None, target_locales: list[str] | None = None
-    ) -> list[tuple[Document, Document, str]]:
-        """Process stale translations using appropriate strategies.
+    def get_missing_translations(
+        self, target_locales: list[str] | None = None, limit: int | None = None
+    ) -> list[tuple[Document, Document | None, str]]:
+        """Find English articles that are missing translations in enabled locales.
+
+        Args:
+            target_locales: List of locales to check (defaults to AI+HYBRID enabled locales)
+            limit: Maximum number of missing translations to return
+        Returns:
+            List of tuples: (english_document, None, target_locale)
+        """
+        target_locales = self._get_target_locales(target_locales)
+
+        missing_translations: list[tuple[Document, Document | None, str]] = []
+
+        for locale in target_locales:
+            if limit is not None and len(missing_translations) >= limit:
+                break
+
+            # Find English docs that don't have a translation in this locale
+            english_docs_without_translation = (
+                self._get_base_english_docs_query()
+                .exclude(translations__locale=locale)
+                .select_related("latest_localizable_revision")
+                .order_by("-latest_localizable_revision__created")
+            )
+
+            # Exclude docs with pending revisions for this locale
+            pending_revisions = Revision.objects.filter(
+                document__parent=OuterRef("pk"),
+                document__locale=locale,
+                creator=self.sumo_bot,
+                is_approved=False,
+                reviewed__isnull=True,
+            )
+
+            docs = (
+                english_docs_without_translation.annotate(has_pending=Exists(pending_revisions))
+                .filter(has_pending=False)
+            )
```

Comment:
I don't think this code is needed.  When there is a pending revision in a locale, there will be a translation document in that locale, so the `.exclude(translations__locale=locale)` above already takes care of this.

Reply:
good point!